patches/server/1045-Add-OfflinePlayer-ifOnline.patch

```diff
@@ -0,0 +1,46 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: leguan <longboard.noah@gmail.com>
+Date: Sun, 28 Jan 2024 19:39:01 +0100
+Subject: [PATCH] Add OfflinePlayer#ifOnline
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java b/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java
+index 4a875bce9563f3b9351ebecde9b0eb1287beb50e..a5ce3a727271b6808fefa4c61989c6e949ecdd5f 100644
+--- a/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java
++++ b/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java
+@@ -358,6 +358,15 @@ public class CraftOfflinePlayer implements OfflinePlayer, ConfigurationSerializa
+         return null;
+     }
+ 
++    // Paper start - Add OfflinePlayer#ifOnline
++    @Override
++    public void ifOnline(final java.util.function.@org.jetbrains.annotations.NotNull Consumer<Player> consumer) {
++        if (this.isOnline()) {
++            consumer.accept(this.getPlayer());
++        }
```

Comment:
We could save some resource by caching `getPlayer` into a variable as `isOnline` invokes `getPlayer` and performs a not-null condition.
```suggestion
        Player player = this.getPlayer();
        if (player != null) {
            consumer.accept(player);
        }
```
and do the same in `CraftPlayer`

Reply:
good point.